cmd/tugger/main.go

```diff
@@ -199,14 +199,19 @@ func mutateAdmissionReviewHandler(w http.ResponseWriter, r *http.Request) {
 
 		// Inject image pull secret
 		if registrySecretName != "" {
+			imagePullSecrets := pod.Spec.ImagePullSecrets
+			if imagePullSecrets == nil {
+				imagePullSecrets = []v1.LocalObjectReference{}
+			}
+			imagePullSecrets = append(imagePullSecrets,
+				v1.LocalObjectReference{
+					Name: registrySecretName,
+				},
+			)
 			patches = append(patches, patch{
 				Op:   "add",
 				Path: "/spec/imagePullSecrets",
```

Comment:
I wonder if this behavior could be achieved more simply with 
```suggestion
				Path: "/spec/imagePullSecrets/-",
```
which is the pseudo-path used to append to the array of imagePullSecrets. What I don't know is if it will work when imagePullSecrets is null.

Reply:
Okay, I tested this. `Path: "/spec/imagePullSecrets/-"` only works when `/spec/imagePullSecrets` is already an array. If it's nil, it fails with `The request is invalid`. So you could use this trick after checking for nil, but it doesn't simplify the logic compared to what you're doing now.